sdk-extensions/jfr-events/src/main/java/io/opentelemetry/sdk/extension/jfr/JfrSpanProcessor.java

```diff
@@ -45,6 +46,9 @@ public boolean isStartRequired() {
 
   @Override
   public void onEnd(ReadableSpan rs) {
+    if (closed) {
```

Comment:
This will leave data in the map as we never touch it again. Since things are weak we won't hold to span structures but we will keep the weak refs around until JVM shutdown.

This was a one reason I went with the NoopMap to avoid the issue to figure out where to place the closed checked :)

probably best to just put it in the existing if statement as the first check.

Reply:
Thanks!